src/crimson/os/seastore/segment_manager/ephemeral.h

```diff
@@ -20,6 +20,15 @@ struct ephemeral_config_t {
   size_t size = 0;
   size_t block_size = 0;
   size_t segment_size = 0;
+
+  void validate() const {
+    ceph_assert_always(size > 0);
+    ceph_assert_always(size <= DEVICE_OFF_MAX);
+    ceph_assert_always(segment_size > 0);
+    ceph_assert_always(segment_size <= MAX_SEG_OFF);
+    ceph_assert_always(size / segment_size > 0);
+    ceph_assert_always(size / segment_size <= DEVICE_SEGMENT_ID_MAX);
+  }
```

Comment:
why do these two validates have different content?

Reply:
They are logically identical to me except that ephemeral uses `ephemeral_config_t` and zns uses `zns_sm_metadata_t`.